### PR TITLE
fixes for multiple rpm issus

### PR DESCRIPTION
**Pull request by erh**
- multiple gofor calls
- overshooting
- "jerk" with lots torque

### CODE REVIEW

File: components/motor/gpio/setup.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ type Config struct {
 	RampRate         float64        `json:"ramp_rate,omitempty"`      // how fast to ramp power to motor when using rpm control
 	MaxRPM           float64        `json:"max_rpm,omitempty"`
 	TicksPerRotation int            `json:"ticks_per_rotation,omitempty"`
+	Debug            bool           `json:"rpm_debug"`
```
Comment (randhid) [inline components/motor/gpio/setup.go:45]:
```suggestion
	Debug            bool           `json:"rpm_debug,omitempty"`
```

File: components/motor/gpio/motor_encoder.go [hunk 2]
```diff
@@ -316,8 +311,15 @@ func (m *EncodedMotor) rpmMonitor() {
...
 
 	rpmSleep, rpmDebug := getRPMSleepDebug()
+	inRamp := false
+
 	for {
-		timer := time.NewTimer(rpmSleep)
+		myRpmSleep := rpmSleep
+		if inRamp {
+			// if we're ramping up or down, make the loop faster
+			myRpmSleep /= 4
```
Comment (randhid) [inline components/motor/gpio/motor_encoder.go:320]: add a todo to finesse pls

File: components/motor/gpio/motor_encoder.go [hunk 3]
```diff
@@ -412,7 +436,15 @@ func (m *EncodedMotor) computeRPM(pos, lastPos, now, lastTime int64) float64 {
 
 func (m *EncodedMotor) computeNewPowerPct(currentRPM, desiredRPM float64) float64 {
 	lastPowerPct := m.state.lastPowerPct
-	if math.Abs(currentRPM) <= 0.001 {
+
+	if desiredRPM*currentRPM < 0 {
+		// if desiredRPM and currentRPM are different signs, we're going the wrong direction
```
Comment (randhid) [inline components/motor/gpio/motor_encoder.go:441]: can you add a to-do to test a negative SetPower Call interrupted by a GoFor in a the opposite direction? (I'm going to look at all these next week)

**Review by randhid — COMMENTED** [review]
A few things/ todos I want added, LGTM otherwise

**Review by randhid — APPROVED** [review]
Left a review with a bunch of small things. DUnno if you want to add tests, but I'll look into that if you'd like.

File: components/motor/gpio/motor_encoder.go (hunk 2 above)
Comment (erh) [reply components/motor/gpio/motor_encoder.go:320]: I generally don't like TODOs unless they are super specific.

File: components/motor/gpio/motor_encoder.go (hunk 3 above)
Comment (erh) [reply components/motor/gpio/motor_encoder.go:441]: i'm going to add some tests soon, don't need to add a TODO
